storage/ndb/rest-server2/extra/drogon/CMakeLists.txt

```diff
@@ -13,57 +13,56 @@
-
-IF(CRYPTO_CUSTOM_LIBRARY) 
-  SET(RDRS_CRYPTO_LIBRARY ${CRYPTO_CUSTOM_LIBRARY})
-ELSE()
-  SET(RDRS_CRYPTO_LIBRARY ${CRYPTO_LIBRARY})
-ENDIF()
+SET(DROGON_VERSION 1.9.7) # See ../README.txt for how to upgrade
 
-IF(OPENSSL_CUSTOM_LIBRARY) 
-  SET(RDRS_OPENSSL_LIBRARY ${OPENSSL_CUSTOM_LIBRARY})
-ELSE()
-  SET(RDRS_OPENSSL_LIBRARY ${OPENSSL_LIBRARY})
-ENDIF()
+set(TRANTOR_USE_TLS "openssl" CACHE STRING "Select TLS library for Trantor" FORCE)
 
-MESSAGE(STATUS "RDRS DROGON RDRS_OPENSSL_LIBRARY ${RDRS_OPENSSL_LIBRARY}")
-MESSAGE(STATUS "RDRS DROGON RDRS_OPENSSL_INCLUDE_DIR ${OPENSSL_INCLUDE_DIR}")
-MESSAGE(STATUS "RDRS DROGON RDRS_CRYPTO_LIBRARY  ${RDRS_CRYPTO_LIBRARY}")
+if(OPENSSL_CUSTOM_LIBRARY)
+    # Point OPENSSL_ROOT_DIR to custom OpenSSL path so Drogon's/trantor's
+    # find_package(OpenSSL) picks them up.
+    set(OPENSSL_ROOT_DIR "${OPENSSL_CUSTOM_LIBRARY}" CACHE PATH "Custom OpenSSL root" FORCE)
+endif()
 
-FILE(WRITE "${RDRS_DROGON_BINARY_DIR}/builder.sh"  "
-  set -e
-  if ! [ -f built ]; then
-    cmake \
-      -S${RDRS_DROGON_SOURCE_DIR} \
-      -B${RDRS_DROGON_BINARY_DIR} \
-      -DBUILD_YAML_CONFIG=OFF \
-      -DCMAKE_INSTALL_PREFIX=${RDRS_DROGON_INSTALL_DIR} \
-      -DRDRS_OPENSSL_LIBRARY=${RDRS_OPENSSL_LIBRARY} \
-      -DRDRS_CRYPTO_LIBRARY=${RDRS_CRYPTO_LIBRARY} \
-      -DRDRS_OPENSSL_INCLUDE_DIR=${OPENSSL_INCLUDE_DIR} \
-      -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
-    make -s -j $(nproc) 
-    make install
-    touch built
-  else
-      echo Skipping Drogon web framework as it is already built. Delete ${RDRS_DROGON_BINARY_DIR}/built to rebuilt it
-  fi"
-)
+# We don't necessarily need these, but we want to catch any compiler errors after patches.
+set(BUILD_EXAMPLES ON CACHE BOOL "Build Drogon examples" FORCE)
+set(BUILD_TESTING ON CACHE BOOL "Enable Drogon tests" FORCE)
 
-EXECUTE_PROCESS(COMMAND bash builder.sh WORKING_DIRECTORY ${RDRS_DROGON_BINARY_DIR} COMMAND_ERROR_IS_FATAL ANY)
-ADD_CUSTOM_TARGET(DROGON_IS_BUILD DEPENDS ${RDRS_DROGON_BINARY_DIR}/built)
+# Add drogon subdirectory
+add_subdirectory(drogon-${DROGON_VERSION})
 
-SET(RDRS_DROGON_CMAKE_MODULES_DIR ${RDRS_DROGON_INSTALL_DIR}/lib/cmake CACHE INTERNAL "Drogon cmake module dir" FORCE)
-SET(RDRS_DROGON_CMAKE_MODULES_DIR_64 ${RDRS_DROGON_INSTALL_DIR}/lib64/cmake CACHE INTERNAL "Drogon cmake module dir" FORCE)
-SET(RDRS_DROGON_INCLUDE_DIR   ${RDRS_DROGON_INSTALL_DIR}/include  CACHE INTERNAL "(Drogon include dir" FORCE)
-SET(RDRS_DROGON_LIB_DIR   ${RDRS_DROGON_INSTALL_DIR}/lib  CACHE INTERNAL "(Drogon lib dir" FORCE)
+# Add include directories (necessary since drogon isn't built standalone)
+target_include_directories(unittest PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/drogon-${DROGON_VERSION}/lib")
 
+# Suppress warnings. This is necessary since RonDB has much stricter compiler
+# flags than drogon.
+#
+# If a build fails due to a warning (raised to an error) in a drogon/trantor
```

Comment:
@smkniazi Does this instruction help to make it compile on thesofa?

Reply:
Instructions doesn't help, how could it ?? You need to fix the build such that it avoids failing when compiling on thesofa by disabling appropriate warning flags. Was the same problem with simdjson 3.12

Reply:
The problem is not specific to the sofa machine. you will have the same problem when building a tarball using docker.